userguide/payment/includes/payment-methods-and-plugins.adoc

```diff
@@ -0,0 +1,14 @@
+=== Payment Methods and plugin names
+
+By default, each plugin has a *name* which is specific to the plugin and configured in the plugin code. On start-up, Kill Bill detects all the plugins and starts them. When a plugin is started, it gets *registered* with Kill Bill with its name. 
+
+When a merchant application invokes the `KillBill#addPaymentMethod`, the name of the plugin corresponding to the payment method needs to be specified. This gets saved in the Kill Bill database as part of the payment method record. 
+
+Kill Bill then uses the plugin name to route the payment processing to the appropriate plugin. So, when it is time to charge the customer against the default payment method, the plugin name corresponding to the payment method is retrieved from the Kill Bill database. Kill Bill then routes the payment to the appropriate plugin. 
```

Comment:
There are 2 ideas here:
* The `plugin_name` is used to *route* the payment operation to the right plugin (payment -> payment method -> plugin_name => plugin)
* Kill Bill also supports a *default payment method*, which is used as a default when processing internal notifications or when user use our apis without specifying a payment method. The default payment method is specified at the Account level (account -> (default) payment method -> plugin_name => plugin)

Perhaps one small paragraph to explain those 2 ideas.

Reply:
Fixed, I've tried to keep it high level and not gone into detail as it would require explaining about the database tables which I dont think makes sense here in the payment userguide.  Please let me know if this is sufficient.